napari/layers/labels/labels.py

```diff
@@ -575,6 +586,25 @@ def color(self, color):
 
         self.color_mode = color_mode
 
+    def _validate_colors(self, labels: Iterable[int]):
+        """Check if after cast to float there is a label collisions"""
+        collision_dkt: Dict[float, int] = {}
+        collision_list = []
+        for label in labels:
+            casted = self._to_vispy_texture_dtype(label)
+            if casted in collision_dkt:
+                collision_list.append((label, collision_dkt[casted]))
+            else:
+                collision_dkt[casted] = label
+        if not collision_list:
+            return
+
+        warn_text = (
+            "Because of the cast to float, the following labels will be merged: "
+            + ", ".join([f"{l1} and {l2}" for l1, l2 in collision_list])
+        )
+        warnings.warn(warn_text, category=RuntimeWarning)
```

Comment:
I think this should be in a different PR tbh... One issue is that the aliasing will happen *both* in direct color mode (where you are adding this check and it's roughly as fast as everything else so not a big deal) and in auto color mode (where engineering a fast solution will be more challenging, especially since new labels can be added at any time). So it feels like a halfway solution.

At any rate, I think both the docstring and the warning should link to #6084, so people can get more details about what's happening. Also, the labels don't get "merged", that sounds like you are messing with their data which is not what's happening.

Finally, I think that pairs are not the right way to present this — for large values, 8, 16, 32 and more labels will get aliased together. So I think they should be presented as lists rather than as pairs.

Anyway, some suggestions:

```suggestion
    def _validate_colors(self, labels: Iterable[int]):
        """Check whether any of the given labels will be aliased together.
        
        See https://github.com/napari/napari/issues/6084 for details.
        """
        aliases: Dict[float, List[int]] = defaultdict(list)
        labels_int = np.fromiter(labels, dtype=int)
        labels_texture = self._to_vispy_texture_dtype(labels_int)
        for integer, texture in zip(labels_int, labels_texture):
            aliases[texture].append(integer)
        aliased_list = [v for k, v in aliases.items() if len(v) > 1]
        if not aliased_list:
            return
        alias_strings = [
            f'Labels {lst[:-1]} will display as the same color as {lst[-1]};'
            for lst in aliased_list
        ]

        warn_text = (
            trans._(
                "Because integer labels are cast to less-precise float for display, "
                "the following label sets will render as the same color:\n"
            )
        ) + '\n'.join(alias_strings)
        warnings.warn(warn_text, category=RuntimeWarning)
```

I forgot how to translate f-strings but that should also be translated.

In the interest of time I suggest this should all get moved to a separate PR because it deals with an orthogonal issue.

Reply:
extracted to #6193